### PR TITLE
Fi 530 revinclude skip

### PR DESCRIPTION
**Pull request by arscan**
If revinclude doesn't return anything, we should skip as it just means they have an incomplete record.  At the very least we should skip if there are no resources of the type you are revincluding against (e.g. if there are no AllergyIntolerances, there should not be any provenances either), but this update should handle both cases.

Pull requests into Inferno require the following items to be completed. Submitter and reviewer 
should check the relevant check boxes when the associated item is done. For items that are not 
applicable, note it's not applicable ("N/A") and check the box. For example, external Pull 
Requests do not require ticket links.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Internal ticket for this PR:
- [x] Internal ticket links to this PR
- [x] Internal ticket is properly labeled (Community/Program)
- [x] Internal ticket has a justification for its Community/Program label
- [ ] Code diff has been reviewed for extraneous/missing code
- [ ] Tests are included and test edge cases
- [x] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: generator/uscore/uscore_generator.rb [hunk 1]
```diff
@@ -217,7 +217,7 @@ def create_revinclude_test(sequence)
                 assert_response_ok(reply)
                 assert_bundle_response(reply)
                 #{resource_variable} = reply&.resource&.entry&.map(&:resource)&.any? { |resource| resource.resourceType == '#{resource_name}' }
-                assert #{resource_variable}, 'No #{resource_name} resources were returned from this search'
+                skip 'No #{resource_name} resources were returned from this search' unless #{resource_variable}
```
Comment (Jammjammjamm) [inline generator/uscore/uscore_generator.rb]: I think this was correct as it was. This change skips the test if no Provenance resources are returned, not if none of the resources the Provenance resources should refer to are found. The search and interaction tests skip using this when no resources have been found:
```
skip 'No resources appear to be available for this patient. Please use patients with more information.' unless @resources_found
```

**Comment by czh-orz** [comment]
It already skips if no resource was found during this line:
`search_params.each { |param, value| skip "Could not resolve #{param} in given resource" if value.nil? }`.

File: generator/uscore/uscore_generator.rb (hunk 1 above)
Comment (arscan) [reply generator/uscore/uscore_generator.rb]: revInclude is weird and a little mind-bendy, but in practice for US Core, using AllergyIntolerance (as an example) we should:
1) skip this test if no AllergyIntolerances exist for the patient.  I agree that skip line you provided would accomplish this, but I'm not seeing that line in the test.  And my update doesn't directly test this, but instead assumes that if you don't have any provenance, then you won't have any AlergyIntolerance, and then it skips.
2) skip this test if no Provenances are linked to any AllergyIntollerances for this patient.  We need to know they support provenances for AllergyIntolerances, so skip if we don't see any.

I could be missing something big, obviously.

**Comment by arscan** [comment]
> It already skips if no resource was found during this line:
> `search_params.each { |param, value| skip "Could not resolve #{param} in given resource" if value.nil? }`.

where is that?  I'm looking at this generated file: https://github.com/onc-healthit/inferno/blob/development/lib/modules/uscore_v3.1.0/us_core_allergyintolerance_sequence.rb#L173

**Comment by czh-orz** [comment]
You can see it sequences where the first search isn't a search by just patient. For example, CarePlan or SmokingStatus.

File: generator/uscore/uscore_generator.rb (hunk 1 above)
Comment (Jammjammjamm) [reply generator/uscore/uscore_generator.rb]: Yeah, I think we need two skips in order for the skips to be clear and not misleading:

- One if no resources of this type have been found at all (could use the line I quoted)
- One if no Provenance resources are found.

Comment (Jammjammjamm) [reply generator/uscore/uscore_generator.rb]: Yeah, I think we need two skips in order for the skips to be clear and not misleading:

- One if no resources of this type have been found at all (could use the line I quoted)
- One if no Provenance resources are found.

**Comment by arscan** [comment]
We have another PR that works in this space #389 and we should just roll this into that update, and closs this.

**Comment by arscan** [comment]
I'm pulling this one in, and then the other PR addresses some of the other comments.
